python/ray/tune/impl/tuner_internal.py

```diff
@@ -151,7 +150,7 @@ def fit(self) -> ResultGrid:
             analysis = self._fit_internal(trainable, param_space)
         else:
             analysis = self._fit_resume(trainable)
-        analysis._legacy_checkpoint = False
+
```

Comment:
Why is this removed - shouldn't Tuner just return the new `Checkpoints`?

Reply:
It's a bit confusing, as ResultGrid is not using `ExperimentAnalysis.get_best_checkpoint` in any way, where `_legacy_checkpoint` is taking effect through. 

It's currently only using `ExperimentAnalysis.best_trial` and then `trial.checkpoint.value` --> convert to a new Checkpoint class.

Reply:
Ok, so to be clear, this means we are still always returning a `Checkpoint` and not a `TrialCheckpoint`, correct? Can we assert that in the test? `assert isinstance(cp, Checkpoint) and not isinstance(cp, TrialCheckpoint)`

Reply:
Yes, that's correct. Added the assertion!

Reply:
I think we have to specifically check `not isinstance(cp, TrialCheckpoint)` as `TrialCheckpoint` is also a `Checkpoint`